crates/sui-types/src/effects/mod.rs

```diff
@@ -423,6 +425,14 @@ pub trait TransactionEffectsAPI {
     fn unsafe_add_object_tombstone_for_testing(&mut self, obj_ref: ObjectRef);
 }
 
+pub struct ObjectChange {
```

Comment:
Open to name changes here to have this clash less with the equivalent concept from Effects V2.

Reply:
We could rename EffectsObjectChange to explicitly let it be known that EOC is only used for v2, and rename ObjectChange to SuiEffectsObjectChange (following SuiSystemStateSummary) maybe

Reply:
I'm not sure that really clarifies much -- we've made the names longer, and different, but the difference in the name doesn't shed light on why these two things are different.  In any case, there wasn't resistance to the naming scheme here, so no need to do anything, I don't think!